src/utilities/sequence-editor/sequence-linter.ts

```diff
@@ -302,141 +304,163 @@ function validateCommandTypeMixing(node: SyntaxNode): Diagnostic[] {
-      if (child.name !== 'Enum' && child.name !== 'Object') {
-        diagnostics.push({
-          from: child.from,
-          message: `@INPUT_PARAMS values are required to be Enums`,
-          severity: 'error',
-          to: child.to,
-        });
-      } else {
-        const variable = {
-          name: text.slice(child.from, child.to),
-          type: 'STRING',
-        } as VariableDeclaration;
-
-        variables.push(variable);
-
-        const metadata: SyntaxNode | null = child?.nextSibling;
-
-        if (metadata !== null) {
-          const properties = metadata.getChildren('Property');
-          let allowableRanges: string | undefined = undefined;
-          let isEnum = false;
-          let isString = false;
-          let enumName: string | undefined = undefined;
-
-          properties.forEach(property => {
-            const propertyNameNode = property.getChild('PropertyName');
-            const propertyValueNode = propertyNameNode?.nextSibling;
-
-            if (propertyNameNode !== null && propertyValueNode !== null && propertyValueNode !== undefined) {
-              const propertyName = text.slice(propertyNameNode.from, propertyNameNode.to);
-              const propertyValue = text.slice(propertyValueNode.from, propertyValueNode.to);
-
-              switch (propertyName.toLowerCase()) {
-                case '"allowable_ranges"':
-                  allowableRanges = propertyValue;
-                  break;
-                case '"enum_name"':
-                  enumName = propertyValue;
-                  break;
-                case '"type"':
-                  isEnum = propertyValue === '"ENUM"';
-                  isString = propertyValue === '"STRING"';
-                  break;
-              }
-            }
-          });
-
-          if (isEnum && enumName === undefined) {
-            diagnostics.push({
-              from: child.from,
-              message: '"enum_name" is required for ENUM type.',
-              severity: 'error',
-              to: child.to,
-            });
-          } else if (!isEnum && enumName !== undefined) {
-            diagnostics.push({
-              from: child.from,
-              message: `"enum_name": ${enumName} is not required for non-ENUM type.`,
-              severity: 'error',
-              to: child.to,
-            });
-          } else if (isString && allowableRanges !== undefined) {
-            diagnostics.push({
-              from: child.from,
-              message: `'allowable_ranges' is not required for STRING type.`,
-              severity: 'error',
-              to: child.to,
-            });
-          }
+function getVariableInfo(
+  parameter: SyntaxNode,
+  text: string,
+): {
+  enumName: string | undefined;
+  name: string | undefined;
+  range: string | undefined;
+  type: string | undefined;
+  values: string | undefined;
+} {
+  const nameNode = parameter.getChild('Enum');
+  const typeNode = parameter.getChild('Type');
```

Comment:
These may be pushed into the if (typeNode) block as they're side effect free and are unused outside of that path
`const enumNode = parameter.getChild('EnumName');
  const rangeNode = parameter.getChild('Range');
  const valuesNode = parameter.getChild('Values');`

Reply:
Addressed with commit 3244803be50f2ee2de696c1459c4dc03d3d4bb87